pkg/filesystem/subvolume.go

```diff
@@ -201,17 +208,100 @@ func unMarshaljson(list string) []fsStruct {
 	if errg != nil {
 		logging.Fatal(errg)
 	}
-
 	return unmarshal
 }
 
 func Delete(ctx context.Context, clientsets *k8sutil.Clientsets, OperatorNamespace, CephClusterNamespace, fs, subvol, svg string) {
 	k8sSubvolume := getK8sRefSubvolume(ctx, clientsets)
 	_, check := k8sSubvolume[subvol]
 	if !check {
-		exec.RunCommandInOperatorPod(ctx, clientsets, "ceph", []string{"fs", "subvolume", "rm", fs, subvol, svg, "--retain-snapshots"}, OperatorNamespace, CephClusterNamespace, true)
+		deleteOmap(ctx, clientsets, OperatorNamespace, CephClusterNamespace, subvol, fs)
+		_, err := exec.RunCommandInOperatorPod(ctx, clientsets, "ceph", []string{"fs", "subvolume", "rm", fs, subvol, svg, "--retain-snapshots"}, OperatorNamespace, CephClusterNamespace, true)
```

Comment:
Why retain snapshots? If we're deleting the stale subvolumes, seems like their snapshots are also not needed.

Reply:
if snapshots are present we can't simply delete the subvolumes, we first need to delete the snapshots and then the subvolume. And I am not very sure above deleting the snapshots from admin side. Hence, best way to delete the subvolume would be with retain-snapshots.
cc @nixpanic  @Madhu-1 what do you think?

Reply:
Ceph-CSI uses a reference counter in OMAP, I think... Only when the reference counter is 0, the subvolume/snapshot should be deleted. Maybe that is only for snapshots, not sure it is used for (plain) subvolumes too.

Reply:
What is the behavior of `--retain-snapshots`? If the subvolume is being deleted, are the snapshots somehow still useful? I thought the point of this feature was to clean up all the stale resources, and snapshots seem stale in this case too.

Reply:
Yes snapshots are meant for backup , even if the subvolume is stale for some reason, the user can restore from the snapshots later